src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -1948,41 +1804,51 @@ static void Main()
 
             verifier.VerifyIL("C.Test3()", @"
 {
-  // Code size       23 (0x17)
+  // Code size       25 (0x19)
   .maxstack  3
-  IL_0000:  ldc.i4.3
+  IL_0000:  ldc.i4.4
   IL_0001:  newarr     ""byte""
   IL_0006:  dup
-  IL_0007:  ldtoken    ""<PrivateImplementationDetails>.__StaticArrayInitTypeSize=3 <PrivateImplementationDetails>.77AF778B51ABD4A3C51C5DDD97204A9C3AE614EBCCB75A606C3B6865AED6744E""
+  IL_0007:  ldtoken    ""int <PrivateImplementationDetails>.F3D4280708A6C4BEA1BAEB5AD5A4B659E705A90BDD448840276EA20CB151BE57""
   IL_000c:  call       ""void System.Runtime.CompilerServices.RuntimeHelpers.InitializeArray(System.Array, System.RuntimeFieldHandle)""
-  IL_0011:  call       ""System.ReadOnlySpan<byte> System.ReadOnlySpan<byte>.op_Implicit(byte[])""
-  IL_0016:  ret
+  IL_0011:  ldc.i4.0
+  IL_0012:  ldc.i4.3
+  IL_0013:  newobj     ""System.ReadOnlySpan<byte>..ctor(byte[], int, int)""
```

Comment:
I am confused about when we do which of the following in various scenarios:
1. create an array, then create a span referencing it (which we do here)
2. create a span referencing a blob (which we do in some of the other tests)

I expected that for UTF-8 string literals, we would only need to do (2) and not (1). Could you please explain?

Reply:
>I am confused about when we do which of the following in various scenarios:
>1. create an array, then create a span referencing it (which we do here)
>2. create a span referencing a blob (which we do in some of the other tests)
>
>I expected that for UTF-8 string literals, we would only need to do (2) and not (1). >Could you please explain?

Utilizing a blob is an optimization strategy that is applied by emit layer. It is applied conditionally, including some conditions that aren't related to specific array data.   
This can be observed in ```TryEmitReadonlySpanAsBlobWrapper``` helper. For example:
- Is ```_module.SupportsPrivateImplClass``` true
- Is ```WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer``` constructor available

In the emit layer we are trying to optimize usage of other (non-pointer) constructors and some conversions. The ```WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer``` is not used in any other compiler layers.

In lowering, we are always using a constructor that takes array, start and length. Emit layer might or might not optimize it with a blob. This specific test covers scenario when ```WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer``` is unavailable. Note ```comp.MakeMemberMissing(WellKnownMember.System_ReadOnlySpan_T__ctor_Pointer);``` above. This disable the optimization and the usage of the regular constructor survives.